docs/se/integrations/neo4j.adoc

```diff
@@ -0,0 +1,343 @@
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j-metrics</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j-health</artifactId>
+</dependency>
+----
+
+Next add the connection configuration properties for Neo4j:
+
+[source,properties]
+----
+neo4j:
+ uri: bolt://localhost:7687
+ authentication:
+   username: neo4j
+   password: secret
+ pool:
+   metricsEnabled: true
+----
+
+This includes both connection information and enables Neo4j metrics propagation.
+
+Finally, we are able to use the `Neo4j` driver.
+
+[source,java]
+----
+@ApplicationScoped
+public class MovieRepository {
+
+    private final Driver driver;
+
+    public MovieRepository(Driver driver) { // <1>
+        this.driver = driver;
+    }
+
+    public List<Movie> findAll() { // <2>
+
+        try (var session = driver.session()) {
+
+            var query = ""
+                    + "match (m:Movie) "
+                    + "match (m) <- [:DIRECTED] - (d:Person) "
+                    + "match (m) <- [r:ACTED_IN] - (a:Person) "
+                    + "return m, collect(d) as directors, collect({name:a.name, roles: r.roles}) as actors";
+
+            return session.readTransaction(tx -> tx.run(query).list(r -> {
+                var movieNode = r.get("m").asNode();
+
+                var directors = r.get("directors").asList(v -> {
+                    var personNode = v.asNode();
+                    return new Person(personNode.get("born").asInt(), personNode.get("name").asString());
+                });
+
+                var actors = r.get("actors").asList(v -> {
+                    return new Actor(v.get("name").asString(), v.get("roles").asList(Value::asString));
+                });
+
+                var m = new Movie(movieNode.get("title").asString(), movieNode.get("tagline").asString());
+                m.setReleased(movieNode.get("released").asInt());
+                m.setDirectorss(directors);
+                m.setActors(actors);
+                return m;
+            }));
+        }
+    }
+}
+
+----
+<1> Constructor with `Neo4j` driver parameter
+<2> Use `Neo4j` driver to extract all Movies
+
+Movies can now be returned as JSON objects:
+
+[source,java]
+----
+public class MovieService implements Service {
+    private final MovieRepository movieRepository;
+
+    public MovieService(MovieRepository movieRepository) {
+        this.movieRepository = movieRepository;
+    }
+
+    @Override
+    public void update(Routing.Rules rules) {
+        rules.get("/api/movies", this::findMoviesHandler);
+    }
+
+    private void findMoviesHandler(ServerRequest request, ServerResponse response) {
+        response.send(this.movieRepository.findAll());
+    }
+}
+
+----
+
+To use the service, as well as to add metrics and health support the following routing should be created:
```

Comment:
To use the service, as well as to add metrics and health support, the following routing should be created:

Reply:
fixed :)